blog/_posts/2020-03-31-extensions.md

```diff
@@ -0,0 +1,52 @@
+---
+slug: more-customizable-tiltfiles-with-extensions
+date: 2020-04-01
+author: victor
+layout: blog
+title: "More customizable Tiltfiles with Extensions"
+subtitle: "Share Tiltfile code snippets with the Tilt community in a streamlined platform. And get a free t-shirt!"
+image: "/assets/images/more-customizable-tiltfiles-with-extensions/michael-dziedzic-XTblNijO9IE-unsplash.jpg"
+image_caption: "Photo by <a href='https://unsplash.com/@lazycreekimages?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText'>Michael Dziedzic</a> on <a href='https://unsplash.com/?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText'>Unsplash</a>"
+tags:
+  - tilt
+keywords:
+  - extensions
+  - tilt
+  - tiltfile
+---
+
+Tilt lets you declare your dev environment in Starlark, a subset of Python. But Starlark is missing a big feature! There's no package manager. Starlark doesn't have built-in ways to share your configuration.
+
+The Tilt community noticed! We've seen people:
+- Copy and paste "best practices" for different languages
+- Struggle to share Tiltfiles across multiple repos
+- Send each other Tiltfile snippets
+
+So we've created Tilt extensions as a first baby step toward a multi-service dev package manager to address this gap.
+
+Tilt extensions are available starting in [v0.12.11](https://github.com/windmilleng/tilt/releases/tag/v0.12.11). [Upgrade](https://docs.tilt.dev/upgrade.html) to the latest version of Tilt.
+
+## What's an extension, exactly? And how do you use it?
+An extension is a packaged function that you import and call, all right inside your Tiltfile. 
+
+For example, to use the [`hello_world` extension](https://github.com/windmilleng/tilt-extensions/tree/master/hello_world), first [`load()`](https://docs.tilt.dev/api.html#api.load) it by including this command in your Tiltfile:
+
+```
+load('ext://hello_world', 'hi')
+```
+
+Then call [`hi()`](https://github.com/windmilleng/tilt-extensions/tree/master/hello_world/Tiltfile) later in your Tiltfile to print "Hello World!".
+
+Read the [docs](https://docs.tilt.dev/extensions.html) for additional details and look at the [tilt-extensions repo](https://github.com/windmilleng/tilt-extensions) for a list of available extensions.
+
+## I want to contribute an extension
+If you're interested in contributing an extension, you just need to know some basic Python syntax. And if you've had practice editing a Tiltfile, that's good enough. Knowledge of Go (which Tilt is written in) is not required in order to contribute. Follow the instructions in [Contribute an Extension](https://docs.tilt.dev/contribute_extension.html) to submit a pull request.
```

Comment:
I'm not a fan of the tone in the first two sentences—"you just need to know..." To me in conveys a false sense of simplicity, when in reality there is finesse involved. Python is simple but writing good Python is not. I suggest rephrasing.

Reply:
I rephrased this so it's more matter-of-fact, and doesn't make subjective statements.